observability/observabilitytest/observabilitytest_test.go

```diff
@@ -12,6 +12,134 @@
+		t.Fatalf("When check recorded values: want nil got %v", err)
+	}
+	if err := observabilitytest.CheckValueViewReceiverDroppedSpans(receiverName, 13); err != nil {
+		t.Fatalf("When check recorded values: want nil got %v", err)
+	}
+	// Test unexpected tag values
+	if err := observabilitytest.CheckValueViewReceiverReceivedSpans(exporterName, 17); err == nil {
+		t.Fatalf("When check recorded values: want not-nil got nil")
+	}
+	if err := observabilitytest.CheckValueViewReceiverDroppedSpans(exporterName, 13); err == nil {
+		t.Fatalf("When check recorded values: want not-nil got nil")
+	}
+	// Test unexpected recorded values
+	if err := observabilitytest.CheckValueViewReceiverReceivedSpans(receiverName, 13); err == nil {
+		t.Fatalf("When check recorded values: want not-nil got nil")
+	}
+	if err := observabilitytest.CheckValueViewReceiverDroppedSpans(receiverName, 17); err == nil {
+		t.Fatalf("When check recorded values: want not-nil got nil")
+	}
+}
+
+func TestCheckValueViewExporterViews(t *testing.T) {
+	defer observabilitytest.SetupRecordedMetricsTest()()
+
+	receiverCtx := observability.ContextWithReceiverName(context.Background(), receiverName)
+	exporterCtx := observability.ContextWithExporterName(receiverCtx, exporterName)
+	observability.RecordTraceExporterMetrics(exporterCtx, 17, 13)
+	// Test expected values.
+	if err := observabilitytest.CheckValueViewExporterReceivedSpans(receiverName, exporterName, 17); err != nil {
+		t.Fatalf("When check recorded values: want nil got %v", err)
+	}
+	if err := observabilitytest.CheckValueViewExporterDroppedSpans(receiverName, exporterName, 13); err != nil {
+		t.Fatalf("When check recorded values: want nil got %v", err)
+	}
+	// Test unexpected tag values
+	if err := observabilitytest.CheckValueViewExporterReceivedSpans(receiverName, receiverName, 17); err == nil {
+		t.Fatalf("When check recorded values: want not-nil got nil")
+	}
+	if err := observabilitytest.CheckValueViewExporterDroppedSpans(receiverName, receiverName, 13); err == nil {
+		t.Fatalf("When check recorded values: want not-nil got nil")
+	}
+	// Test unexpected recorded values
+	if err := observabilitytest.CheckValueViewExporterReceivedSpans(receiverName, exporterName, 13); err == nil {
+		t.Fatalf("When check recorded values: want not-nil got nil")
+	}
+	if err := observabilitytest.CheckValueViewExporterDroppedSpans(receiverName, exporterName, 17); err == nil {
+		t.Fatalf("When check recorded values: want not-nil got nil")
+	}
+}
+
+func TestNoSetupCalled(t *testing.T) {
+	receiverCtx := observability.ContextWithReceiverName(context.Background(), receiverName)
+	observability.RecordTraceReceiverMetrics(receiverCtx, 17, 13)
+	// Failed to check because views are not registered.
+	if err := observabilitytest.CheckValueViewReceiverReceivedSpans(receiverName, 17); err == nil {
+		t.Fatalf("When check recorded values: want not-nil got nil")
+	}
+}
+
+func TestCheckRecordedMetricsForTraceExporter_WithNoMetricsRecorded(t *testing.T) {
```

Comment:
I assume there will be another PR on tests on MetricsExporter?

Reply:
Yes.